src/sentry/eventstore/models.py

```diff
@@ -356,17 +356,13 @@ def get_hashes(self, force_config: StrategyConfiguration | None = None) -> Calcu
         # Create fresh hashes
         from sentry.grouping.api import sort_grouping_variants
 
-        variants = self.get_grouping_variants(force_config)
-        flat_variants = sort_grouping_variants(variants)
-        flat_hashes = self._hashes_from_sorted_grouping_variants(flat_variants)
-
-        if flat_hashes:
-            sentry_sdk.set_tag("get_hashes.flat_variant", flat_hashes[0])
-
-        flat_hashes_values = [hash_ for _, hash_ in flat_hashes]
+        _variants = self.get_grouping_variants(force_config)
+        variants = sort_grouping_variants(_variants)
+        hashes = self._hashes_from_sorted_grouping_variants(variants)
+        hashes_values = [hash_ for _, hash_ in hashes]
 
         return CalculatedHashes(
-            hashes=flat_hashes_values,
+            hashes=hashes_values,
             variants=variants,
         )
```

Comment:
Perhaps better suited to a follow-up PR to this one, but `_hashes_from_sorted_grouping_variants` is only ever used here AFAICT, and we ignore the first item in each of the tuples it returns - meaning we could simplify it to just return the hash values, and then this

```python
        hashes = self._hashes_from_sorted_grouping_variants(variants)
        hashes_values = [hash_ for _, hash_ in hashes]

        return CalculatedHashes(
            hashes=hashes_values,
            variants=variants,
        )
```

could be replaced with just

```python
        return CalculatedHashes(
            hashes=self._hashes_from_sorted_grouping_variants(variants),
            variants=variants,
        )
```

or (if I'm reading it right) even just

```python
        return CalculatedHashes(
            hashes={ variant.get_hash() for variant in variants.values() } - { None },
            variants=variants,
        )
```

(eliminating `_hashes_from_sorted_grouping_variants` entirely).